syschdemd.sh

```diff
@@ -40,4 +40,10 @@ if [[ $# -gt 0 ]]; then
 else
     cmd="$userShell"
 fi
-exec $sw/nsenter -t $(< /run/systemd.pid) -p -m -- $sw/machinectl -q --uid=@defaultUser@ shell .host /bin/sh -c "cd \"$PWD\"; exec $cmd"
+
+# Pass external environment but filter variables specific to root user.
+exportCmd="$(export -p | $sw/grep -vE ' (HOME|LOGNAME|SHELL|USER)='); export WSLPATH=\"$PATH\""
+
+exec $sw/nsenter -t $(< /run/systemd.pid) -p -m -- $sw/machinectl -q \
+	--uid=@defaultUser@ shell .host /bin/sh -c \
+	"cd \"$PWD\"; $exportCmd; exec $cmd"
```

Comment:
```suggestion
exec $sw/nsenter -t $(< /run/systemd.pid) -p -m --wd="$PWD" -- $sw/env WSLPATH="$PATH" @wrapperDir@/su -s $userShell @defaultUser@ "$@"
```

I'm just passing through `PATH` as `WSLPATH` and using your change to `configuration.nix`. Works really well.

I'm not sure why you have the rest, I think other environment variables are already passed through:

```
$ env | grep WSL
WSL_INTEROP=/run/WSL/807_interop
WSL_DISTRO_NAME=NixOS
WSLENV=WT_SESSION::WT_PROFILE_ID
```

So a simple `env` seems like enough.

Reply:
`machinectl` is needed to properly start a user session, and it doesn't pass through the environment on its own. By replacing the call to `machinectl`, `systemd` won't start user services, for example.